packages/blob/src/index.ts

```diff
@@ -215,8 +223,42 @@ export async function list(
   };
 }
 
-function getApiUrl(): string {
-  return process.env.VERCEL_BLOB_API_URL || 'https://blob.vercel-storage.com';
+export interface GenerateUploadUrlOptions extends BlobCommandOptions {
+  pathname: string;
+  onUploadCompletedUrl: string;
+}
+
+export function generateUploadTokenFromReadWriteToken({
+  pathname,
+  onUploadCompletedUrl,
+  token,
+}: GenerateUploadUrlOptions): string {
+  const timestamp = new Date();
+  timestamp.setSeconds(timestamp.getSeconds() + 30);
```

Comment:
sure 2 mins is fine as well. I considered that the token needs to be valid for the time the client makes a request to CF and CF calls the vercel validation endpoint, which should be pretty low. Increasing the token time would increase the risk of multiple requests, though these would be quite useless as the user can upload only a file with the same pathname

Reply:
I am thinking about a page that contains a `<form>` with many fields, some of them being file uploads (going directly to Cloudflare with our technique). Probably the token would be generated once at page load and that's it. In this case we should probably have a default token duration that is long enough (and even 2 minutes may not be sufficient).

That, or we should document how to write a useClientUploadToken that relies on SWR to revalidate the token when time has passed (this may be the best option overall).

Reply:
Oh right, multiple files I did not even think about that (soon enough we will need prefix: option for generate blob token to solve that easily, I'll make a note of it)

Reply:
> even if you have multiple files in the same form, I'd always couple the token generation to the blob.put request. Otherwise you cannot say for how much time a token should be valid

You're right, even more since the token will be used once, let's make sure our default example in README does a submitHandler + generateToken + uploadFile then (correct?).

Reply:
Yep I think so for now it's fine